src/SolverAPI.jl

```diff
@@ -174,26 +174,38 @@ gracefully and included in `Response`.
     the `MOI.ModelLike` model and should return `Nothing`.
   - `request`: A `SolverAPI.Request` specifying the optimization problem.
   - `solver`: A `MOI.AbstractOptimizer` to use to solve the problem.
+
+## Keyword Args
+
+  - `bridge_type`: The type to use for the solver cache and bridge. Either `Int` or `Float64`. Defaults to `Float64`.
 """
-function solve(fn, json::Request, solver::MOI.AbstractOptimizer)
+function solve(
+    fn,
+    json::Request,
+    solver::MOI.AbstractOptimizer;
+    bridge_type::Type = Float64,
+)
     errors = validate(json)
     isempty(errors) || return response(; errors)
 
-    # TODO (dba) `SolverAPI.jl` should be decoupled from any solver specific code.
     solver_info = Dict{Symbol,Any}()
-    if lowercase(get(json.options, :solver, "highs")) == "minizinc"
-        T = Int
+    if bridge_type == Int
         solver_info[:use_indicator] = false
```

Comment:
We also ultimately want this `solver_info` dict to get passed in like the model type. In future there will likely be more things than just `:use_indicator`, and it is actually unrelated to the bridge_type

Reply:
maybe actually the bridge_type/numeric_type should just be put inside the solver_info dict. that would mean passing it around less

Reply:
Yeah, that sounds like a good idea!

Reply:
I've decided to instead of passing solver info as a `Dict` to use keyword arguments instead. I think that is more user-friendly.

Reply:
That's ok. In future we may have a lot more of these sorts of keyword args, to control how models are built depending on which solver is used. That's why I introduced the solver_info dict originally.